In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("coretech_clients_cleaned.csv")

In [3]:
df.head()

,client_id,client_name,company_name,service,project_budget,project_duration_days,client_city,lead_source,status,rating
0,C001,John Smith,TechNova,Web Development,15000,45,New York,LinkedIn,Completed,4.8
1,C002,Sarah Johnson,DataWave,Data Analytics,22000,60,Chicago,Website,In Progress,4.5
2,C003,Michael Brown,CloudNet,Cloud Migration,35000,90,Houston,Referral,Completed,4.9
3,C004,Emily Davis,InnoSoft,Mobile App Development,28000,75,Los Angeles,LinkedIn,Pending,4.3
4,C005,David Wilson,SmartSys,Cybersecurity,18000,40,Seattle,Website,Completed,4.7


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   client_id              30 non-null     object 
 1   client_name            30 non-null     object 
 2   company_name           30 non-null     object 
 3   service                30 non-null     object 
 4   project_budget         30 non-null     int64  
 5   project_duration_days  30 non-null     int64  
 6   client_city            30 non-null     object 
 7   lead_source            30 non-null     object 
 8   status                 30 non-null     object 
 9   rating                 30 non-null     float64
dtypes: float64(1), int64(2), object(7)
memory usage: 2.5+ KB


In [5]:
df.describe()

,project_budget,project_duration_days,rating
count,30.000000,30.000000,30.000000
mean,26783.333333,71.066667,4.596667
std,12031.340778,32.144001,0.256614
min,9500.000000,22.000000,4.100000
25%,17250.000000,45.000000,4.400000
50%,25500.000000,69.000000,4.600000
75%,34250.000000,88.750000,4.800000
max,52000.000000,140.000000,5.000000


In [7]:
#Question 1: Highest Demand Service
df["service"].value_counts()

service
Data Analytics            5
Cloud Migration           5
Cybersecurity             5
Web Development           4
Mobile App Development    4
AI Solutions              4
Software Testing          3
Name: count, dtype: int64

Highest Demand Service: Data Analytics

In [9]:
#Question 2: City With Most Clients
df["client_city"].value_counts()

client_city
New York        1
Chicago         1
Cleveland       1
New Orleans     1
Kansas City     1
Milwaukee       1
Baltimore       1
Memphis         1
Jacksonville    1
Columbus        1
Indianapolis    1
Nashville       1
Charlotte       1
Philadelphia    1
Detroit         1
Las Vegas       1
San Jose        1
Portland        1
Austin          1
San Diego       1
Phoenix         1
Atlanta         1
Dallas          1
Miami           1
Denver          1
Boston          1
Seattle         1
Los Angeles     1
Houston         1
Raleigh         1
Name: count, dtype: int64

All cities have equal distribution

In [12]:
#Question 3: Best Lead Source
df["lead_source"].value_counts()

lead_source
Website         9
Referral        9
LinkedIn        7
Social Media    5
Name: count, dtype: int64

Best Lead Source: Website

In [14]:
#Question 4: Average Project Budget
df["project_budget"].mean()

26783.333333333332

Average Budget: $26,783.33

In [18]:
#Question 5: Highest Budget Service
df.groupby("service")['project_budget'].mean().sort_values(ascending=False)

service
AI Solutions              48500.000000
Cloud Migration           35200.000000
Mobile App Development    30000.000000
Data Analytics            23000.000000
Cybersecurity             22000.000000
Web Development           14500.000000
Software Testing          10166.666667
Name: project_budget, dtype: float64

Highest Budget Service: AI Solutions

In [21]:
#Question 6: Project Status Counts
df["status"].value_counts()

status
Completed      19
In Progress     7
Pending         4
Name: count, dtype: int64

In [23]:
#Question 7: Average Client Rating
df["rating"].mean()

4.596666666666667

Average Rating: 4.6

#Question 8: Insights

1. Data Analytics is the most demanded service.
2. AI Solutions generate the highest revenue.
3. Website is the strongest lead source.
4. CoreTech has a strong completion rate.
5. Client satisfaction is high.

#Question 9: Recommendations

1. Expand AI service offerings.
2. Improve website marketing.
3. Scale Data Analytics services.